libslide/src/evaluator_rules/registry.rs

```diff
@@ -55,18 +55,31 @@ impl RuleSet {
     /// Creates a list of `Rules`s from the unbuilt rule set.
     pub fn build(&self) -> Vec<Rule> {
         let num_rules = self.rules.values().fold(0, |sum, ur| match ur {
-            UnbuiltRule::M(m) => sum + m.len(),
+            // Building a string rule actually generates two versions:
+            // 1. The "raw" form of the string rule
+            // 2. A version of the (1) boostrapped with a set of rules, possibly including (1)
+            //    itself. This allows a rule to match in a number of evaluated contexts that cannot
+            //    be represented by a string rule. For example, the rule
+            //
+            //      -(_a - _b) -> _b - _a
+            //
+            //    in its raw form may only be applied to an expression with explicit parentheses.
+            //    By bootstrapping this rule with a rule that removes explicit parentheses, the
+            //    rule can be applied on expressions with implicit parentheses (i.e. of the prefix
+            //    form `(- (- _a _b))`).
+            UnbuiltRule::S(_) => sum + 2,
             _ => sum + 1,
```

Comment:
doesn't this limit our form of rules to just 2 for m set rules. Is there a case where theres more then 2 forms?

Reply:
I got rid of M rules